File: ui/InsuranceUploadScreen.kt
package com.example.healthride.ui

import android.Manifest
import android.content.pm.PackageManager
import android.net.Uri
import android.util.Log
import androidx.activity.compose.rememberLauncherForActivityResult
import androidx.activity.result.contract.ActivityResultContracts
import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowBack
import androidx.compose.material.icons.filled.CameraAlt
import androidx.compose.material.icons.filled.FileUpload
import androidx.compose.material.icons.outlined.CameraAlt
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.core.content.ContextCompat
import androidx.core.content.FileProvider
import coil.compose.rememberAsyncImagePainter
import com.example.healthride.ui.theme.*
import java.io.File
import java.text.SimpleDateFormat
import java.util.*
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.verticalScroll
import androidx.lifecycle.viewmodel.compose.viewModel
import com.example.healthride.ui.viewmodel.UserViewModel
import androidx.compose.runtime.livedata.observeAsState


@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun InsuranceUploadScreen(
    userViewModel: UserViewModel = viewModel(),
    onBackClick: () -> Unit = {},
    onVerificationSubmitted: () -> Unit = {}
) {
    // --- State Observation ---
    val insuranceCardUrl by userViewModel.insuranceCardUrl.observeAsState()
    val idCardUrl by userViewModel.idCardUrl.observeAsState()
    val isLoading by userViewModel.isLoading.observeAsState(false)
    val error by userViewModel.error.observeAsState()
    val verificationSubmitted by userViewModel.verificationSubmitted.observeAsState(false)

    // --- Local UI State ---
    var uriForCamera by remember { mutableStateOf<Uri?>(null) }
    var currentUploadTarget by remember { mutableStateOf<String?>(null) } // "insurance" or "id"

    val context = LocalContext.current
    val packageName = context.packageName

    // --- Permissions ---
    var hasCameraPermission by remember {
        mutableStateOf( ContextCompat.checkSelfPermission( context, Manifest.permission.CAMERA ) == PackageManager.PERMISSION_GRANTED )
    }
    val requestPermissionLauncher = rememberLauncherForActivityResult(
        contract = ActivityResultContracts.RequestPermission()
    ) { isGranted ->
        if (isGranted) {
            hasCameraPermission = true
            uriForCamera?.let { uri ->
                Log.d("InsuranceUpload", "Permission granted, launching camera for $currentUploadTarget")
                // NOTE: cameraLauncher must be defined *before* this point for this to work immediately
                // If cameraLauncher is defined below, this immediate launch might fail.
                // Consider moving launcher definitions up if immediate launch after permission is needed.
                // For now, just set hasCameraPermission = true is sufficient.
            }
        } else {
            Log.w("InsuranceUpload", "Camera permission denied.")
        }
    }

    // --- File Creation ---
    fun createImageFileUri(): Uri? {
        return try {
            val timeStamp = SimpleDateFormat("yyyyMMdd_HHmmss", Locale.US).format(Date())
            val imageFileName = "JPEG_${timeStamp}_"
            val storageDir: File? = context.externalCacheDir ?: context.cacheDir
            if (storageDir == null) {
                Log.e("InsuranceUpload", "Failed to get cache directory.")
                return null
            }
            val imageFile = File.createTempFile(imageFileName, ".jpg", storageDir)
            FileProvider.getUriForFile(context, "$packageName.fileprovider", imageFile)
        } catch (ex: Exception) {
            Log.e("InsuranceUpload", "Error creating image file URI: ${ex.message}", ex)
            null
        }
    }

    // --- Activity Result Launchers (Defined BEFORE they are used in permission callback if needed) ---
    val cameraLauncher = rememberLauncherForActivityResult(
        contract = ActivityResultContracts.TakePicture()
    ) { success: Boolean ->
        val target = currentUploadTarget
        val uri = uriForCamera // Use the stored URI
        currentUploadTarget = null
        uriForCamera = null // Reset after use

        if (success && uri != null && target != null) {
            Log.d("InsuranceUpload", "$target camera success. URI: $uri")
            if (target == "insurance") userViewModel.uploadInsuranceCard(uri)
            else if (target == "id") userViewModel.uploadIdCard(uri)
        } else {
            Log.w("InsuranceUpload", "$target camera failed or URI/target was null. Success: $success")
        }
    }

    val galleryLauncher = rememberLauncherForActivityResult(
        contract = ActivityResultContracts.GetContent()
    ) { uri: Uri? ->
        val target = currentUploadTarget
        currentUploadTarget = null

        if (uri != null && target != null) {
            Log.d("InsuranceUpload", "$target gallery selected. URI: $uri")
            if (target == "insurance") userViewModel.uploadInsuranceCard(uri)
            else if (target == "id") userViewModel.uploadIdCard(uri)
        } else {
            Log.w("InsuranceUpload", "$target gallery selection cancelled or URI/target was null.")
        }
    }

    // --- Side Effects ---
    LaunchedEffect(verificationSubmitted) {
        if (verificationSubmitted == true) {
            onVerificationSubmitted()
        }
    }
    LaunchedEffect(Unit) {
        userViewModel.loadCurrentUser()
    }

    // --- UI ---
    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Insurance Verification") },
                navigationIcon = { IconButton(onClick = onBackClick) { Icon( Icons.Default.ArrowBack, "Back" ) } },
                colors = TopAppBarDefaults.topAppBarColors( containerColor = SurfaceWhite, titleContentColor = TextDarkBlue )
            )
        }
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .background(BackgroundGray)
                .padding(paddingValues)
                .padding(horizontal = 24.dp, vertical = 16.dp)
                .verticalScroll(rememberScrollState()),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Text( "Upload Your Documents", style = MaterialTheme.typography.headlineSmall, fontWeight = FontWeight.Bold, color = TextDarkBlue )
            Spacer(Modifier.height(8.dp))
            Text( "We need clear photos of your Insurance Card and a valid ID.", style = MaterialTheme.typography.bodyMedium, color = TextMediumGray, textAlign = TextAlign.Center )
            Spacer(Modifier.height(24.dp))

            // Error Display
            if (error != null) {
                Text(
                    text = "Error: $error", // Display the error message from ViewModel
                    color = MaterialTheme.colorScheme.error,
                    style = MaterialTheme.typography.bodySmall,
                    modifier = Modifier.padding(bottom = 16.dp).fillMaxWidth(),
                    textAlign = TextAlign.Center
                )
            }

            // Insurance Card Section - Calling DocumentUploadCard
            DocumentUploadCard(
                title = "Insurance Card",
                imageUrl = insuranceCardUrl,
                // Show loading if general loading is true AND target was insurance
                isLoading = isLoading == true && currentUploadTarget == "insurance",
                onTakePhotoClick = {
                    currentUploadTarget = "insurance" // Set target
                    if (hasCameraPermission) {
                        createImageFileUri()?.let { uri ->
                            uriForCamera = uri // Store URI
                            Log.d("InsuranceUpload", "Launching camera for insurance. Target URI: $uri")
                            cameraLauncher.launch(uri) // Launch camera
                        } ?: Log.e("InsuranceUpload", "Failed to create URI for insurance photo")
                    } else {
                        Log.d("InsuranceUpload", "Requesting camera permission for insurance.")
                        uriForCamera = createImageFileUri() // Prepare URI in case permission granted
                        requestPermissionLauncher.launch(Manifest.permission.CAMERA) // Request permission
                    }
                },
                onUploadClick = {
                    currentUploadTarget = "insurance"; // Set target
                    Log.d("InsuranceUpload", "Launching gallery for insurance.")
                    galleryLauncher.launch("image/*") // Launch gallery
                }
            )

            Spacer(Modifier.height(16.dp))

            // Identification Card Section - Calling DocumentUploadCard
            DocumentUploadCard(
                title = "Identification Card",
                description = "(Driver's License or Government ID)",
                imageUrl = idCardUrl,
                // Show loading if general loading is true AND target was id
                isLoading = isLoading == true && currentUploadTarget == "id",
                onTakePhotoClick = {
                    currentUploadTarget = "id" // Set target
                    if (hasCameraPermission) {
                        createImageFileUri()?.let { uri ->
                            uriForCamera = uri // Store URI
                            Log.d("InsuranceUpload", "Launching camera for ID. Target URI: $uri")
                            cameraLauncher.launch(uri) // Launch camera
                        } ?: Log.e("InsuranceUpload", "Failed to create URI for ID photo")
                    } else {
                        Log.d("InsuranceUpload", "Requesting camera permission for ID.")
                        uriForCamera = createImageFileUri() // Prepare URI
                        requestPermissionLauncher.launch(Manifest.permission.CAMERA) // Request permission
                    }
                },
                onUploadClick = {
                    currentUploadTarget = "id"; // Set target
                    Log.d("InsuranceUpload", "Launching gallery for ID.")
                    galleryLauncher.launch("image/*") // Launch gallery
                }
            )

            Spacer(Modifier.height(24.dp))

            // Submit Button
            Button(
                onClick = {
                    Log.d("InsuranceUpload", "Submit button clicked.")
                    userViewModel.submitForVerification()
                },
                modifier = Modifier.fillMaxWidth().height(56.dp),
                shape = RoundedCornerShape(28.dp),
                colors = ButtonDefaults.buttonColors(
                    containerColor = PrimaryBlue,
                    disabledContainerColor = PrimaryBlue.copy(alpha = 0.5f)
                ),
                // Enable only if both URLs present AND not currently loading generally
                enabled = !insuranceCardUrl.isNullOrBlank() && !idCardUrl.isNullOrBlank() && isLoading != true
            ) {
                // Show loading if general loading is true AND no specific target (i.e., submitting)
                if (isLoading == true && currentUploadTarget == null) {
                    CircularProgressIndicator( color = Color.White, strokeWidth = 2.dp, modifier = Modifier.size(24.dp) )
                } else {
                    Text( "Submit for Verification", style = MaterialTheme.typography.titleMedium, fontWeight = FontWeight.Bold )
                }
            }
            Spacer(Modifier.height(16.dp))
        }
    }
}


/**
 * Reusable Composable for the document upload card UI.
 * This function MUST be defined within the same file or accessible scope
 * where it's called by InsuranceUploadScreen.
 */
@Composable
private fun DocumentUploadCard(
    title: String,
    description: String? = null,
    imageUrl: String?, // URL of the uploaded image (placeholder or real)
    isLoading: Boolean, // Is an upload specifically for this card in progress?
    onTakePhotoClick: () -> Unit,
    onUploadClick: () -> Unit
) {
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(containerColor = SurfaceWhite),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
    ) {
        Column(
            modifier = Modifier.padding(16.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            // Title and optional Description
            Text(title, style = MaterialTheme.typography.titleMedium, fontWeight = FontWeight.Bold, color = TextDarkBlue)
            if (description != null) {
                Spacer(Modifier.height(4.dp))
                Text(description, style = MaterialTheme.typography.bodySmall, color = TextMediumGray, textAlign = TextAlign.Center)
            }
            Spacer(Modifier.height(16.dp))

            // Image Display Area
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(200.dp)
                    .clip(RoundedCornerShape(8.dp))
                    .background(BackgroundGray)
                    .border(1.dp, TextLightGray, RoundedCornerShape(8.dp)),
                contentAlignment = Alignment.Center
            ) {
                // Show loading indicator if this specific card's upload is happening
                if (isLoading) {
                    CircularProgressIndicator(color = PrimaryBlue)
                }
                // Show image if URL is present AND not currently loading this card
                else if (!imageUrl.isNullOrBlank()) {
                    Image(
                        painter = rememberAsyncImagePainter(imageUrl), // Use Coil to load image from URL
                        contentDescription = title,
                        modifier = Modifier.fillMaxSize(),
                        contentScale = ContentScale.Fit // Fit ensures the whole image is visible
                    )
                }
                // Show placeholder if no image and not loading
                else {
                    Column(horizontalAlignment = Alignment.CenterHorizontally) {
                        Icon(Icons.Default.CameraAlt, null, Modifier.size(48.dp), tint = TextLightGray)
                        Spacer(Modifier.height(8.dp))
                        Text("Take photo or upload", color = TextMediumGray)
                    }
                }
            } // End Box

            Spacer(Modifier.height(16.dp))

            // Action Buttons Row
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceEvenly
            ) {
                val isImagePresent = !imageUrl.isNullOrBlank()

                // Take Photo / Retake Button
                Button(
                    onClick = onTakePhotoClick,
                    modifier = Modifier.weight(1f),
                    shape = RoundedCornerShape(12.dp),
                    colors = if (isImagePresent) ButtonDefaults.outlinedButtonColors(contentColor = PrimaryBlue)
                    else ButtonDefaults.buttonColors(containerColor = PrimaryBlue),
                    border = if (isImagePresent) BorderStroke(1.dp, PrimaryBlue) else null,
                    enabled = !isLoading // Disable only if this card is loading
                ) {
                    Icon(
                        imageVector = if (isImagePresent) Icons.Outlined.CameraAlt else Icons.Default.CameraAlt,
                        contentDescription = null,
                        modifier = Modifier.size(18.dp)
                    )
                    Spacer(Modifier.width(8.dp))
                    Text(if (isImagePresent) "Retake" else "Take Photo")
                }

                Spacer(Modifier.width(8.dp))

                // Upload / Upload New Button
                OutlinedButton(
                    onClick = onUploadClick,
                    modifier = Modifier.weight(1f),
                    shape = RoundedCornerShape(12.dp),
                    border = BorderStroke(1.dp, PrimaryBlue),
                    enabled = !isLoading // Disable only if this card is loading
                ) {
                    Icon(
                        imageVector = Icons.Default.FileUpload,
                        contentDescription = null,
                        tint = PrimaryBlue,
                        modifier = Modifier.size(18.dp)
                    )
                    Spacer(Modifier.width(8.dp))
                    Text(if (isImagePresent) "Upload New" else "Upload")
                }
            } // End Row
        } // End Column
    } // End Card
} // End DocumentUploadCard

File: lib/src/screens/insurance_upload_screen.dart
import 'dart:io';
import 'package:flutter/material.dart';
import 'package:image_picker/image_picker.dart';
import '../constants/app_color.dart';
import '../routes/app_routes.dart';

/// Screen for uploading insurance and identification documents
class InsuranceUploadScreen extends StatefulWidget {
  /// Creates an insurance upload screen
  const InsuranceUploadScreen({super.key});

  @override
  State<InsuranceUploadScreen> createState() => _InsuranceUploadScreenState();
}

class _InsuranceUploadScreenState extends State<InsuranceUploadScreen> {
  // Mock service for demo purposes
  String? _insuranceCardUrl;
  String? _idCardUrl;
  bool _isLoading = false;
  String? _error;
  String? _currentUploadTarget;

  @override
  void initState() {
    super.initState();
    _loadUserData();
  }

  Future<void> _loadUserData() async {
    // Mock implementation
    // In a real app, you would fetch this from a service
    setState(() {
      _insuranceCardUrl = null;
      _idCardUrl = null;
    });
  }

  Future<void> _takePicture(String documentType) async {
    setState(() {
      _currentUploadTarget = documentType;
      _isLoading = true;
      _error = null;
    });

    try {
      final ImagePicker picker = ImagePicker();
      final XFile? image = await picker.pickImage(source: ImageSource.camera);

      if (image != null) {
        final File imageFile = File(image.path);
        await _uploadImage(imageFile, documentType);
      }
    } catch (e) {
      setState(() {
        _error = 'Failed to take picture: $e';
      });
    } finally {
      setState(() {
        _isLoading = false;
      });
    }
  }

  Future<void> _uploadFromGallery(String documentType) async {
    setState(() {
      _currentUploadTarget = documentType;
      _isLoading = true;
      _error = null;
    });

    try {
      final ImagePicker picker = ImagePicker();
      final XFile? image = await picker.pickImage(source: ImageSource.gallery);

      if (image != null) {
        final File imageFile = File(image.path);
        await _uploadImage(imageFile, documentType);
      }
    } catch (e) {
      setState(() {
        _error = 'Failed to upload from gallery: $e';
      });
    } finally {
      setState(() {
        _isLoading = false;
      });
    }
  }

  Future<void> _uploadImage(File imageFile, String documentType) async {
    // Mock implementation
    await Future.delayed(const Duration(seconds: 2));

    setState(() {
      if (documentType == 'insurance') {
        _insuranceCardUrl = 'https://example.com/mock-insurance-card.jpg';
      } else if (documentType == 'id') {
        _idCardUrl = 'https://example.com/mock-id-card.jpg';
      }
    });
  }

  Future<void> _submitForVerification() async {
    setState(() {
      _currentUploadTarget = null;
      _isLoading = true;
      _error = null;
    });

    try {
      // Mock submission
      await Future.delayed(const Duration(seconds: 2));

      if (mounted) {
        Navigator.of(context)
            .pushReplacementNamed(AppRoutes.verificationStatus);
      }
    } catch (e) {
      setState(() {
        _error = 'Failed to submit for verification: $e';
      });
    } finally {
      setState(() {
        _isLoading = false;
      });
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Insurance Verification'),
        backgroundColor: Colors.transparent,
        elevation: 0,
      ),
      body: SafeArea(
        child: SingleChildScrollView(
          padding: const EdgeInsets.all(24.0),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.center,
            children: [
              const Text(
                'Upload Your Documents',
                style: TextStyle(
                  fontSize: 24,
                  fontWeight: FontWeight.bold,
                  color: AppColor.textDarkBlue,
                ),
              ),
              const SizedBox(height: 8),
              const Text(
                'We need clear photos of your Insurance Card and a valid ID.',
                style: TextStyle(
                  color: AppColor.textMediumGray,
                ),
                textAlign: TextAlign.center,
              ),
              const SizedBox(height: 24),

              // Error message
              if (_error != null)
                Container(
                  width: double.infinity,
                  padding: const EdgeInsets.all(12),
                  margin: const EdgeInsets.only(bottom: 16),
                  decoration: BoxDecoration(
                    color: AppColor.errorRedLight,
                    borderRadius: BorderRadius.circular(8),
                  ),
                  child: Text(
                    _error!,
                    style: const TextStyle(
                      color: AppColor.errorRed,
                      fontSize: 14,
                    ),
                    textAlign: TextAlign.center,
                  ),
                ),

              // Insurance Card Upload
              _DocumentUploadCard(
                title: 'Insurance Card',
                imageUrl: _insuranceCardUrl,
                isLoading: _isLoading && _currentUploadTarget == 'insurance',
                onTakePhotoClick: () => _takePicture('insurance'),
                onUploadClick: () => _uploadFromGallery('insurance'),
              ),

              const SizedBox(height: 16),

              // ID Card Upload
              _DocumentUploadCard(
                title: 'Identification Card',
                description: "(Driver's License or Government ID)",
                imageUrl: _idCardUrl,
                isLoading: _isLoading && _currentUploadTarget == 'id',
                onTakePhotoClick: () => _takePicture('id'),
                onUploadClick: () => _uploadFromGallery('id'),
              ),

              const SizedBox(height: 24),

              // Submit button
              SizedBox(
                width: double.infinity,
                height: 56,
                child: ElevatedButton(
                  onPressed: (_insuranceCardUrl != null &&
                          _idCardUrl != null &&
                          !_isLoading)
                      ? _submitForVerification
                      : null,
                  style: ElevatedButton.styleFrom(
                    backgroundColor: AppColor.primaryBlue,
                    foregroundColor: Colors.white,
                    shape: RoundedRectangleBorder(
                      borderRadius: BorderRadius.circular(28),
                    ),
                    elevation: 0,
                  ),
                  child: _isLoading && _currentUploadTarget == null
                      ? const CircularProgressIndicator(
                          color: Colors.white,
                          strokeWidth: 2,
                        )
                      : const Text(
                          'Submit for Verification',
                          style: TextStyle(
                            fontSize: 16,
                            fontWeight: FontWeight.bold,
                          ),
                        ),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

/// A card for document uploads
class _DocumentUploadCard extends StatelessWidget {
  final String title;
  final String? description;
  final String? imageUrl;
  final bool isLoading;
  final VoidCallback onTakePhotoClick;
  final VoidCallback onUploadClick;

  const _DocumentUploadCard({
    required this.title,
    this.description,
    this.imageUrl,
    required this.isLoading,
    required this.onTakePhotoClick,
    required this.onUploadClick,
  });

  @override
  Widget build(BuildContext context) {
    return Container(
      width: double.infinity,
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(16),
        boxShadow: [
          BoxShadow(
            color: Colors.black.withOpacity(0.05),
            blurRadius: 10,
            offset: const Offset(0, 2),
          ),
        ],
      ),
      child: Padding(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          children: [
            // Title and description
            Text(
              title,
              style: const TextStyle(
                fontSize: 18,
                fontWeight: FontWeight.bold,
                color: AppColor.textDarkBlue,
              ),
            ),
            if (description != null)
              Padding(
                padding: const EdgeInsets.only(top: 4),
                child: Text(
                  description!,
                  style: const TextStyle(
                    fontSize: 14,
                    color: AppColor.textMediumGray,
                  ),
                  textAlign: TextAlign.center,
                ),
              ),
            const SizedBox(height: 16),

            // Image display area
            Container(
              width: double.infinity,
              height: 200,
              decoration: BoxDecoration(
                color: AppColor.backgroundGray,
                borderRadius: BorderRadius.circular(8),
                border: Border.all(
                  color: AppColor.textLightGray,
                  width: 1,
                ),
              ),
              child: isLoading
                  ? const Center(
                      child: CircularProgressIndicator(
                        color: AppColor.primaryBlue,
                      ),
                    )
                  : imageUrl != null
                      ? ClipRRect(
                          borderRadius: BorderRadius.circular(8),
                          child: Image.network(
                            imageUrl!,
                            fit: BoxFit.contain,
                            errorBuilder: (context, error, stackTrace) {
                              return const Center(
                                child: Text(
                                  'Error loading image',
                                  style: TextStyle(color: AppColor.errorRed),
                                ),
                              );
                            },
                          ),
                        )
                      : Column(
                          mainAxisAlignment: MainAxisAlignment.center,
                          children: const [
                            Icon(
                              Icons.camera_alt,
                              size: 48,
                              color: AppColor.textLightGray,
                            ),
                            SizedBox(height: 8),
                            Text(
                              'Take photo or upload',
                              style: TextStyle(
                                color: AppColor.textMediumGray,
                              ),
                            ),
                          ],
                        ),
            ),
            const SizedBox(height: 16),

            // Buttons
            Row(
              children: [
                // Take Photo Button
                Expanded(
                  child: ElevatedButton.icon(
                    onPressed: isLoading ? null : onTakePhotoClick,
                    icon: const Icon(
                      Icons.camera_alt,
                      size: 18,
                    ),
                    label: Text(imageUrl != null ? 'Retake' : 'Take Photo'),
                    style: ElevatedButton.styleFrom(
                      backgroundColor: imageUrl != null
                          ? Colors.white
                          : AppColor.primaryBlue,
                      foregroundColor: imageUrl != null
                          ? AppColor.primaryBlue
                          : Colors.white,
                      side: imageUrl != null
                          ? const BorderSide(color: AppColor.primaryBlue)
                          : null,
                      shape: RoundedRectangleBorder(
                        borderRadius: BorderRadius.circular(12),
                      ),
                    ),
                  ),
                ),
                const SizedBox(width: 8),
                // Upload Button
                Expanded(
                  child: OutlinedButton.icon(
                    onPressed: isLoading ? null : onUploadClick,
                    icon: const Icon(
                      Icons.file_upload,
                      size: 18,
                    ),
                    label: Text(imageUrl != null ? 'Upload New' : 'Upload'),
                    style: OutlinedButton.styleFrom(
                      side: const BorderSide(color: AppColor.primaryBlue),
                      shape: RoundedRectangleBorder(
                        borderRadius: BorderRadius.circular(12),
                      ),
                    ),
                  ),
                ),
              ],
            ),
          ],
        ),
      ),
    );
  }
}
